solr/core/src/java/org/apache/solr/query/FilterQuery.java

```diff
@@ -89,13 +87,13 @@ public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float bo
 
     if (!(searcher instanceof SolrIndexSearcher)) {
       // delete-by-query won't have SolrIndexSearcher
-      return new BoostQuery(new ConstantScoreQuery(q), 0).createWeight(searcher, scoreMode, 1f);
+      return new ConstantScoreQuery(q).createWeight(searcher, scoreMode, 1f);
     }
 
     SolrIndexSearcher solrSearcher = (SolrIndexSearcher)searcher;
     DocSet docs = solrSearcher.getDocSet(q);
     // reqInfo.addCloseHook(docs);  // needed for off-heap refcounting
 
-    return new BoostQuery(new SolrConstantScoreQuery(docs.getTopFilter()), 0).createWeight(searcher, scoreMode, 1f);
+    return docs.makeQuery().createWeight(searcher, scoreMode, 1f);
```

Comment:
I'm confused what is going on here. We take the query and get the docs and then use the docs to make a query to get a constant score weight. This seems wasteful but maybe there is some nuance in the processing that I'm missing.

Reply:
The call to `solrSearcher.getDocSet(q)` is the point of all this -- which is cached.